### PR TITLE
storage: make it cheaper to clone Scheduler

### LINKED ISSUE #11500: Cloning the scheduler wastes much CPU
Running a TiDB sysbench write only workload on a recent nightly TiKV, I it uses about 5% total CPU to clone and drop the inner `Engine` (the inner `RaftKv`):
image

`RaftKv` is a rather complex struct. And we actually don't need to clone them because it can be fetched from the TLS of scheduler threads.

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve?

Issue Number: closes #11500

### What is changed and how it works?

This PR removes the engine from `Scheulder`. This should reduce some CPU usage of the scheduler threads.

And it also moves `control_mutex` into `SchedulerInner`, so we can avoid some clones when it is not used.

### Check List 

Tests 

- Existing tests.

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by youjiali1995 — APPROVED** [review]
Good point. I missed it after always getting snapshot in the scheduler pool.

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -283,10 +286,10 @@ impl<L: LockManager> SchedulerInner<L> {
 /// Scheduler which schedules the execution of `storage::Command`s.
 #[derive(Clone)]
 pub struct Scheduler<E: Engine, L: LockManager> {
-    // `engine` is `None` means currently the program is in scheduler worker threads.
-    engine: Option<E>,
```
Comment (BusyJay) [inline src/storage/txn/scheduler.rs:287]: It doesn't seem to be used at all.

**Review by ekexium — APPROVED** [review]

File: src/storage/txn/scheduler.rs (hunk 1 above)
Comment (sticnarf) [reply src/storage/txn/scheduler.rs:287]: Right. But we need to keep the type, so `with_tls_engine` can know what the actual type is.

**Comment by BusyJay** [comment]
How come the CI of my PRs almost green while this one almost red?

**Pull request merged**